### PR TITLE
add Ollama Embedding class

### PR DESCRIPTION
**Pull request by chnsagitchen**
# Description

Add Ollama embedding class for generate embedding for query or document in local with Ollama

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] I stared at the code and made sure it makes sense
- [ ] I test  the implementation with my local Ollama service

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/llms/ollama_embedding.py
Comment (logan-markewich) [inline llama_index/llms/ollama_embedding.py:1]: Thanks for contributing this! Can you move this file to `llama_index/embeddings` instead?

File: llama_index/llms/ollama_embedding.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                "Please install requests with `pip install requests`"
+            )
+
+        ollama_request_body = {
+            "prompt": prompt,
+            "model": self.model_name,
+            **self.ollama_additional_kwargs,
+        }
+
+        response = requests.post(
```
Comment (logan-markewich) [inline llama_index/llms/ollama_embedding.py:103]: since this is just requests, we can use `httpx` to define proper async methods as well? 🙏🏻

**Review by logan-markewich — COMMENTED** [review]
Thanks for the contribution! Just two quick comments. 

If you could add an example notebook in `docs/examples/embeddings` as well, that would be perfect 👍🏻

File: llama_index/llms/ollama_embedding.py
Comment (chnsagitchen) [reply llama_index/llms/ollama_embedding.py:1]: Thanks! I moved it to llama_index/embeddings

**Comment by chnsagitchen** [comment]
> Thanks for the contribution! Just two quick comments.
> 
> If you could add an example notebook in `docs/examples/embeddings` as well, that would be perfect 👍🏻

@logan-markewich just add the example in the following commit.

File: llama_index/llms/__init__.py [hunk 2]
```diff
@@ -75,6 +76,7 @@
     "MockLLM",
     "MonsterLLM",
     "Ollama",
+    "OllamaEmbedding",
```
[the remark this answers is not in the document]
Comment (chnsagitchen) [reply llama_index/llms/__init__.py]: Refactor this by moving ollama_embedding.py to llama_index/embeddings and import it there accordingly.

**Comment by jerryjliu** [comment]
@chnsagitchen while i have you here, this is completely unrelated but can you DM me your twitter handle/linkedin etc. (if you have one), planning to promote the ollama query engine soon

**Review by hatianzhang — APPROVED** [review]
Awesome! LGTM

**Pull request merged**